contrib/ray/raycluster_example.yaml

```diff
@@ -1,60 +1,112 @@
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-ray-workers-head
+spec:
+  action: ALLOW
+  rules:
+  - to:
+    - operation:
+        ports:
+        - "6379"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    ray.io/headless-worker-svc: raycluster-istio
+  name: raycluster-istio-headless-svc
+spec:
+  clusterIP: None
+  selector:
+    ray.io/cluster: kubeflow-raycluster
+  publishNotReadyAddresses: true
+  ports:
+  - name: node-manager-port
+    port: 6380
+    appProtocol: grpc
+  - name: object-manager-port
+    port: 6381
+    appProtocol: grpc
+  - name: runtime-env-agent-port
+    port: 6382
+    appProtocol: grpc
+  - name: dashboard-agent-grpc-port
+    port: 6383
+    appProtocol: grpc
+  - name: dashboard-agent-listen-port
+    port: 52365
+    appProtocol: http
+  - name: metrics-export-port
+    port: 8080
+    appProtocol: http
+  - name: p10002
+    port: 10002
+    appProtocol: grpc
+  - name: p10003
+    port: 10003
+    appProtocol: grpc
+  - name: p10004
+    port: 10004
+    appProtocol: grpc
+  - name: p10005
+    port: 10005
+    appProtocol: grpc
+  - name: p10006
+    port: 10006
+    appProtocol: grpc
+  - name: p10007
+    port: 10007
+    appProtocol: grpc
+  - name: p10008
+    port: 10008
+    appProtocol: grpc
+  - name: p10009
+    port: 10009
+    appProtocol: grpc
+  - name: p10010
+    port: 10010
+    appProtocol: grpc
+  - name: p10011
+    port: 10011
+    appProtocol: grpc
+  - name: p10012
+    port: 10012
+    appProtocol: grpc
```

Comment:
Why do you need so many ports and why is only one in the authorizationpolicy?

Reply:
--node-manager-port: Raylet port for node manager.
--object-manager-port: Raylet port for object manager.
--runtime-env-agent-port: Raylet port for runtime env agent.
--dashboard-agent-grpc-port: The port to listen for grpc on
--max-worker-port: Maximum port number worker can be bound to.
--metrics-export-port: The port to use to expose Ray metrics.

These ports are needed when using the Ray Cluster. The access is given for the relevant ports through the authorization policy